client/client.go

```diff
@@ -1933,19 +1932,23 @@ func (c *Client) allocSync() {
 			}
 
 			var resp structs.GenericResponse
-			if err := c.RPC("Node.UpdateAlloc", &args, &resp); err != nil {
+			err := c.RPC("Node.UpdateAlloc", &args, &resp)
+			if err != nil {
+				// Error updating allocations, do *not* clear
+				// updates and retry after backoff
 				c.logger.Error("error updating allocations", "error", err)
-				syncTicker.Stop()
-				syncTicker = time.NewTicker(c.retryIntv(allocSyncRetryIntv))
-				staggered = true
-			} else {
-				updates = make(map[string]*structs.Allocation)
-				if staggered {
-					syncTicker.Stop()
-					syncTicker = time.NewTicker(allocSyncIntv)
-					staggered = false
-				}
+				syncTicker.Reset(c.retryIntv(allocSyncRetryIntv))
```

Comment:
What is the functional difference between using `Reset` here vs `Stop` and re-creating the ticker, as was done here before, and is still done below?

Reply:
Oh this is a bug! Thanks. Fixing. Resetting a ticker doesn't necessarily clear the chan if a tick has already happened, so this could fail to have it's desired affect.